### PR TITLE
Ensure the results screen shows the last game the player played

### PR DESCRIPTION
**Pull request by mmdonaldson**
Fix for bug #146

### CODE REVIEW

File: application/app.py [hunk 1]
```diff
@@ -302,7 +302,7 @@ def game_result():
     """Endpoint for ending a game mid game
     Inputs - player_id
     Outputs - status comment"""
-    player = Player.query.join(User).filter_by(email=g.current_user["email"]).join(Game).filter_by(complete=True).first()
+    player = Player.query.join(User).filter_by(email=g.current_user["email"]).join(Game).order_by(Game.id.desc()).filter_by(complete=True).first()
```
Comment (jacobstringer) [inline application/app.py:305]: Joining player with both game and user seems like quite an inefficient query - I hadn't noticed when this first got introduced. Does the front end not keep the player id after ending the game - if it did, you could use:

player = get_player(player_id)

Which only queries the player table. The alternative is just to order by Player.id.desc() and get rid of joining game, as SQLAlchemy guarantees that the newest player ids will be the latest games.

I think this 3 way join means the object you get back isn't a player object but rather an Alchemy hybrid, which I think is okay below (as get_players just looks at the player's game id, which would be in the hybrid as well), but I wouldn't be sure of it not causing bugs if we accidentally mistook it for a player and used one of the player methods on it.

**Review by jacobstringer — APPROVED** [review]

File: application/app.py (hunk 1 above)
Comment (mmdonaldson) [reply application/app.py:305]: In terms of efficiency it would definitely be preferable to come from the frontend except that the results page is currently accessible regardless of whether or not you came from a game. We could change this if we wanted and if player id wasn't passed to the results component we could redirect to home.

Changing the query to be the latest player object wouldn't work in case of the situation where the player is in a more recent game that is not complete.

So basically if we make it so the results page must come at the end of a game then this would fix both issues. What do you think @jacobstringer

Comment (jacobstringer) [reply application/app.py:305]: I forgot about needing the game object for the complete tag. That makes more sense now. I hadn't realised that you guys were using this extra API for the end game results. I thought you guys were going to use the game status API for it, so it's all set up to be used for the final status information. As it currently stands, the status returns the game object (so you can see if the game is complete), along with all the players' information.

The only difference between this endpoint and the status endpoint is that in this one, you can supply the email for the result, whereas in the status you need a player id. I suppose it doesn't really make much of a difference in the grand scheme of things, it was more just a case of me realising that we had duplicated work a bit.

Comment (mmdonaldson) [reply application/app.py:305]: Yeah i suppose your proposed approach is just more complicated on the frontend and I am lazy.

**Comment by mmdonaldson** [comment]
@jacobstringer merging but we can discuss tonight
